app/src/main/java/ch/hikemate/app/model/route/ListOfHikeRoutesViewModel.kt

```diff
@@ -51,6 +51,26 @@ open class ListOfHikeRoutesViewModel(
   private suspend fun getRoutesAsync(onSuccess: () -> Unit = {}, onFailure: () -> Unit = {}) {
     withContext(dispatcher) {
       val area = area_.value ?: return@withContext
+
+      // Check if the area is on the date line
+      if (area.lonEast < area.lonWest) {
+        val bounds1 = BoundingBox(area.latNorth, 180.0, area.latSouth, area.lonWest)
+        val bounds2 = BoundingBox(area.latNorth, area.lonEast, area.latSouth, -180.0)
```

Comment:
Suggestion: move this logic to an extension function as well.

```kotlin
fun BoundingBox.splitByDateLine(): Pair<BoundingBox, BoundingBox> {
    return if (isCrossingDateLine()) {
        Pair(
            BoundingBox(latNorth, 180.0, latSouth, lonWest),
            BoundingBox(latNorth, lonEast, latSouth, -180.0)
        )
    }
    else {
       Pair(this, this)
    }
}
```

Then, use it like in Scala (very satisfying) :

```kotlin
val (bounds1, bounds2) = area.splitByDateLine()
```

Reply:
nice!